experimentation/src/test/java/com/automattic/android/experimentation/local/FileBasedCacheTest.kt

```diff
@@ -59,8 +60,23 @@ internal class FileBasedCacheTest {
         sut.clear()
     }
 
-    private fun fileBasedCache(scope: TestScope) = FileBasedCache(
-        cacheDir = createTempDirectory().toFile(),
+    @Test
+    fun `saving cache when cache dir doesnt exist is successful`() = runTest {
+        val cacheDir = File("build/non-existent").apply { assert(!this.exists()) }
+        val sut = fileBasedCache(this, cacheDir = cacheDir)
+        sut.saveAssignments(TEST_ASSIGNMENTS)
+
+        val result = sut.getAssignments()
+
+        assertEquals(TEST_ASSIGNMENTS, result)
+        cacheDir.deleteRecursively()
```

Comment:
It might be safer to use the [`TemporaryFolder`](https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html) rule, like this:

```kotlin
@get:Rule val tempDir = TemporaryFolder()

@Test
fun test() {
  val cacheDir = File(tempDir.newFolder(), "cache")
}
```

This way, you won't have to manually clean up the test directory with  `cacheDir.deleteRecursively()`. If the test fails, the cleanup might not execute, which could cause subsequent test failures due to the existence of `cacheDir.`

Reply:
You make a good point of manual clean up, but using `tempDir.newFolder` defats the purpose of the test, because it will _create_ a new folder (and we want to verify the case of non-existing folder).

I propose a change using setup/teardown methods: f4513966c9c3fbe88960cdeed1b59d7ae9f1750f

Reply:
`val cacheDir = File(tempDir.newFolder(), "cache")` this doesn't create the `cacheDir`. Only the parent folder.

Reply:
Ah right, thanks 👍 addressed in 4bb70bc8764b955ce07b3b660f6c038a255cdea5